build/docker-entrypoint.sh

```diff
@@ -5,13 +5,23 @@
 
 set -o pipefail
 shopt -s dotglob
+# shellcheck disable=SC1091
+source /usr/local/bin/common.sh
 
 function shutdown() {
+  # Watch the fluent-bit process
+  tail --pid="$(pidof fluent-bit)" -f /dev/null &
+
+  # Sleep for twice the service flush interval from fluent-bit.conf
+  sleep .2s
+
   # Have fluent-bit dequeue and then exit
   kill -SIGTERM "$(pidof fluent-bit)"
```

Comment:
@JonZeolla will fluent bit always be done flushing within `.2s`? If not, does fluent-bit ignore the sigterm until it  is done flushing or just shuts down gracefully regardless of what it is doing?

Reply:
I asked the `fluent-bit` development team on December 7th and haven't heard back, although I did get an ack that they were going to look at my question.  [Here](https://fluent-all.slack.com/archives/C0CTQGHKJ/p1670442975403489) is the slack post

Reply:
I did some more testing and it seems like the reason why the logs sometimes don't make it to the remote system is because it isn't pulled in by the `tail` input plugin, not because of dequeueing properly during shutdown.  However, it's still just a theory, and it seems like using the 2x the flush interval is the best way to approach this (as opposed to an arbitrary `sleep` time).  Technically it would just need any number > 1x the flush interval if I'm right. This is based off a conversation I had with a `fluent-bit` developer at KubeCon, but they weren't actually reviewing our configs/exact scenario so I took it with a grain of salt.

Reply:
@JonZeolla sounds good, interested in fluent-bit's response

Reply:
Me too.  I say we ship because I can't repo with the latest config, but if they say there's a better way to do this I'll do a follow on PR